### PR TITLE
QuickSort Code in Kotlin

### PR DESCRIPTION
**Pull request by PriyanshuChatterjee**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the contributing guidelines.

### Description

Quicksort is based on the partitioning of an array of data into smaller arrays. A large array is partitioned into two arrays one of which holds values smaller than the specified value, say pivot, based on which the partition is made and another array holds values greater than the pivot value. Quicksort partitions an array and then calls itself recursively twice to sort the two resulting subarrays. This algorithm is quite efficient for large-sized data sets as its average and worst-case complexity are O(n2), respectively.

### Checklist

- [ x ] I've read the contribution guidelines.
- [ x ] I've checked the issue list before deciding what to submit.
- [ x ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I created an issue. The link of the issue is: #5348

### CODE REVIEW

**Comment by PriyanshuChatterjee** [comment]
@HarshCasper Please review this code.

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
+//Kotlin program to implement QuickSort using recursive approach
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt]: Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    val equalto = items.filter { it == pivot }
+    //Variable lesser stores the list of element with indexes less than the pivot index
+    val lesser = items.filter { it < pivot }
+    //Variable greater stores the list of element with indexes less than the pivot index
+    val greater = items.filter { it > pivot }
+    //Calling the quicksort function recursively on the splitted arrays
+    //This will get recursively called until left with multiple arrays with a single element and arranged in order
+    return quicksort(lesser) + equalto + quicksort(greater)
+}
+//Main Driver Code
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt:33]: add a new line above comments

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Enter the List of numbers you want to Sort:
+23
+55
+22
+90
+67
+54
+23
+Original List: 23	55	22	90	67	
+Ordered list: [22, 23, 23, 54, 55, 67, 90]
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt]: Remove extra test case and Add Time and Space Complexity at the end of the Code

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Update the readme as well

**Comment by PriyanshuChatterjee** [comment]
@ankitaggarwal23 All the changes are done. Please Review it.

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
+//QuickSort Algortihm works on Divide and Conquer Algorithm. It creates two empty arrays to hold elements less than the pivot value and elements greater than the pivot value, and then recursively sort the sub arrays. There are two basic operations in the algorithm, swapping items in place and partitioning a section of the array.
+
+//Here QuickSort is implemented in Kotlin using recursive approach
+
+//Importing the Java.util package, it is needed because we are using the scanner object.
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt]: remove such redundant comments 
- Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Enter the length of the array
+6
+Enter the List of numbers you want to Sort:
+34
+33
+67
+3
+2
+45
+Original List: 34	33	67	3	2
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt]: 45 missing in original list

File: Kotlin/sort/QuickSort/src/QuickSort.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+33
+67
+3
+2
+45
+Original List: 34	33	67	3	2	
+Ordered list: [2, 3, 33, 34, 45, 67]
+
+Time Complexity: O(n log n)[Average]
+Space Complexity: O(log n)[Average]
```
Comment (ankitaggarwal23) [inline Kotlin/sort/QuickSort/src/QuickSort.kt]: remove the average

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Do the changes mentioned

**Comment by PriyanshuChatterjee** [comment]
@ankitaggarwal23 Changes Done.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Readme missing

**Comment by PriyanshuChatterjee** [comment]
@ankitaggarwal23 Added the Readme. Sorry didn't get you in the first go.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**